gui/system/forms.py

```diff
@@ -1705,9 +1705,15 @@ def __init__(self, *args, **kwargs):
         self.instance._original_sys_pool = self.instance.sys_pool
         self.instance._original_sys_syslog_usedataset = self.instance.sys_syslog_usedataset
         self.instance._original_sys_rrd_usedataset = self.instance.sys_rrd_usedataset
-        self.fields['sys_pool'].widget.attrs['onChange'] = (
-            "systemDatasetMigration();"
-        )
+        with client as c:
+            if c.call('system.is_freenas'):
+                self.fields['sys_pool'].widget.attrs['onChange'] = (
+                    "systemDatasetMigration();"
+                )
+            else:
+                self.fields['sys_pool'].widget.attrs['onChange'] = (
+                    "systemDatasetMigration_TN();"
```

Comment:
You probably also want to check for failover licensed so it doesnt show for single nodes?